account_invoice_report_grouped_by_picking_hide_pickings/models/account_move.py

```diff
@@ -10,6 +10,39 @@ class AccountMove(models.Model):
     def lines_grouped_by_picking(self):
         res = super().lines_grouped_by_picking()
         to_remove = list(filter(lambda g: g["picking"].hide_on_invoice, res))
-        for i in to_remove:
-            res.remove(i)
+        to_update = {}
+        if to_remove:
+            if to_remove == res:
+                return []
+            for line in to_remove:
```

Comment:
Estás iterando 2 veces sobre res: aquí y en la línea 12. Yo quitaría lo de la línea 12, iteraría directamente sobre res y haría un break en caso que no se cumpliera g["picking"].hide_on_invoice.

Reply:
De hecho, en la linea 22 vuelvo a iterar sobre res. Tener la variable to_remove me ayuda a tener ya las lineas filtradas y añadirme mas comprobaciones. 

Creo que se puede mejorar el código, pero eso creo que esta bien

Reply:
Is this resolved?